mitiq/lre/lre.py

```diff
@@ -25,6 +25,7 @@ def execute_with_lre(
     executor: Union[Executor, Callable[[QPROGRAM], float]],
```

Comment:
Return type of `Callable` should be generalized to `QuantumResult` similar to to how it exists in ZNE: https://github.com/unitaryfund/mitiq/blob/ff15f44b8bdc57bc6f4e7feef14dc3d3f2b7f86d/mitiq/zne/zne.py#L66